File: algorithms/solutions/python3/0001_two_sum.py

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(nums)):
            another = target - nums[i]
            if another in m:
                return [m[another] , i]
            m[nums[i]] = i
        return [-1, -1]
# @lc code=end

File: algorithms/solutions/java/_0001_TwoSum.java
package hot100;

import java.util.HashMap;
import java.util.Map;

public class _0001_TwoSum {
// @lc code=start
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            Map<Integer, Integer> map = new HashMap<>();
            for (int i = 0; i < nums.length; i++) {
                int another = target - nums[i];
                if (map.containKey(another)){
                    return new int[]{map.get(another), i};
                }
                map.put(nums[i, i]);
            }
            return new int[]{-1, -1};
        }
    }
// @lc code=end
}
